db/migrate/20160310170333_add_service_ancestry.rb

```diff
@@ -0,0 +1,43 @@
+class AddServiceAncestry < ActiveRecord::Migration[5.0]
+  class Service < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  def update_service_parent(parent)
+    Service.where(:service_id => parent.try(:id)).each do |svc|
+      ancestry = [parent.try(:ancestry), parent.try(:id)].compact.join("/")
+      svc.update_attributes(:ancestry => ancestry) if parent
+      update_service_parent(svc)
+    end
+  end
+
+  def up
+    add_column :services, :ancestry, :string
+    add_index :services, :ancestry
+
+    Service.connection.schema_cache.clear!
+    Service.reset_column_information
```

Comment:
This should not be necessary.  If it is, I'm very surprised and it's something that @lpichler should put on his TODO list.